src/openrct2/Game.cpp

```diff
@@ -481,6 +483,33 @@ static void FixInvalidSurfaces()
     }
 }
 
+static void FixRideData()
+{
+    // Fix some ride data which can influence things like the ride's upkeep cost.
+    for (auto& ride : GetRideManager())
+    {
+        const auto& rtd = ride.GetRideTypeDescriptor();
+
+        // Invalid amount of drops also influence the upkeep cost.
+        if (!rtd.HasFlag(RIDE_TYPE_FLAG_HAS_DROPS) && ride.drops != 0)
+        {
+            ride.drops = 0;
+        }
+
+        // Flat rides do not have a segment length, this influences upkeep cost.
+        if (rtd.HasFlag(RIDE_TYPE_FLAG_IS_SHOP_OR_FACILITY) || rtd.HasFlag(RIDE_TYPE_FLAG_FLAT_RIDE))
```

Comment:
I think the assumption about flat rides here might be wrong, some ride rating tests seem to fail, or the assumption is correct and the ride rating tests always used the left over memory.